design-proposals/crd-based-management.md

```diff
@@ -0,0 +1,281 @@
+# CRD-based management
+
+**Authors**: @elevran, @oro
+
+**Begin Design Discussion**: 2023-11-22
+
+**(optional) Status:** draft
+
+## Summary/Abstract
+
+This design proposal explores the design space for managing ClusterLink through
+ Kubernetes (k8s) Custom Resource Definitions (CRDs). It aims to replace the
+ current API and CLI with k8s native management philosophy (e.g., asynchronous
+ reconciliation of actual and desired states) and tooling (e.g., `kubectl`, RBAC).
+ This document focuses on single cluster management. A central management solution
+ could be considered in a future extension. In the context of the current design,
+ information needed by a cluster regarding remote peers (e.g., their gateways,
+ services they expose, etc.) ise assumed to materialize by the user out of thin
+ air/magic (TODO reword... Or it may make sense to have a short paragraph upfront
+ describing what an imaginary CM may provide?).
+
+## Background
+
+### Motivation and problem space
+
+Currently ClusterLink exposes a RESTful API for its users. A convenience CLI (`gwctl`)
+ is provided for easier API access. The API/CLI is based on Created/Read/Update/Delete
+ (CRUD) of the following objects:
+
+- Remote `Peer` clusters this cluster can connect to
+- Service `Export`s this cluster makes available for consumption to remote clusters
+- Service `Import`s that this cluster makes avaialble to local workloads (TODO: `Binding`
+ objects are needed? Are they fundamental to the mental model?)
+- `Policy` objects governing aspects of the cross cluster communication.
+
+An additional CLI (`cl-adm`) is provided to ease the bootstrap and deployment of ClusterLink
+ in a k8s cluster. The `cl-adm` utility creates a set of (mostly k8s configuration) files on
+ the local file systems which can be used to ...
+
+TODO Lack of users and permissions, need to support tooling with relatively low value.
+  cl-adm and gwctl are not consistent.
+
+### Impact and desired outcome
+
+TODO The reverse of some of the above...
+
+### Prior discussion and links
+
+TODO Some initial ideas in GH issue. Reference.
+
+## (Optional) User/User Story
+
+### Users
```

Comment:
Maybe it is worthwhile to be explicit in defining the management objects (CRD) that are proposed and what they are defined. Or, using the current concepts, refer to an existing doc. After an initial read, I do not think you propose to carry over all current management objects from `gwctl`, but I could be wrong.
Consider having a separate section for "runtime configuration concepts", that would include such things as Peer, Export, Policy, ...